embedded/sql/sql_tx_options.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2022 Codenotary Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sql
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/codenotary/immudb/embedded/store"
+)
+
+type TxOptions struct {
+	ReadOnly                bool
+	SnapshotMustIncludeTxID func(lastPrecommittedTxID uint64) uint64
+	SnapshotRenewalPeriod   time.Duration
+}
+
+func DefaultTxOptions() *TxOptions {
+	txOpts := store.DefaultTxOptions()
+
+	return &TxOptions{
+		ReadOnly:                txOpts.Mode == store.ReadOnlyTx,
+		SnapshotMustIncludeTxID: txOpts.SnapshotMustIncludeTxID,
```

Comment:
This sounds like a boolean variable, but it is not, could we have something specific?

Reply:
ReadOnly is in fact a boolean variable